server/sonar-server/src/main/java/org/sonar/server/rule/RegisterRules.java

```diff
@@ -100,9 +104,11 @@ public void start() {
         }
       }
       List<RuleDto> activeRules = processRemainingDbRules(allRules.values(), session);
-      removeActiveRulesOnStillExistingRepositories(session, activeRules, context);
+      List<ActiveRuleChange> changes = removeActiveRulesOnStillExistingRepositories(session, activeRules, context);
       session.commit();
       ruleIndexer.setEnabled(true).index();
+      activeRuleIndexer.setEnabled(true);
```

Comment:
How ?
Indexing can only be done after the commit, and a lot of things can be done before commiting